docs/enhancements/on-prem-data-collection.md

```diff
@@ -0,0 +1,220 @@
+
+1. Implementing new data types to collect is out of the scope of this epic. This epic defines how and when the data already collected will be processed and sent.
+2. Define how SaaS should redesign their data pipeline (taken care of in [MGMT-11244](https://issues.redhat.com/browse/MGMT-11244)) is out of the scope of this epic. This epic only refers to on-premises deployments of AI.
+3. Define the creation and/or management of an event streaming instance such as Kafka and the endpoint this data will be hosted at (both taken care of in [MGMT-11244](https://issues.redhat.com/browse/MGMT-11244)).
+4. Collecting must-gather data (and logs) for debugging failed cluster installation is outside the scope of this epic. This epic is only concerned with [event](https://github.com/openshift/assisted-service/blob/6b8401d3b0f0ceea5a837f33c17b9b1cd5ec52a3/internal/events/api/event.go) data.
+5. Data formatting and handling by the event streaming service is out of the scope of this epic. The event streaming service should be able to process the data from all versions of the on-premises environment.
+
+
+## Proposal
+
+### Data Sending Flow
+
+All of the event data for this cluster will be queried from the DB, packaged, and sent to the data hosting endpoint.
+
+Data will be sent when:
+1. A cluster installation attempt finishes (regardless of success or failure)
+
+Before the data is sent, Assisted Service will get a remote configuration from an endpoint we specify.
+
+Sending data should not interfere with customer workloads and deployment
+
+### Assisted-Service Changes 
+
+Changes that need to be made in the `Assisted-Service` to support this:
+- Needs to know what the data hosting service endpoint is and how to authenticate into it
+- Needs code changes when an install is attempted to query and send data
+- Needs to know how to format the data to send to the data hosting service
+    - Includes sanitizing the data such as hashing user's emails like in the [event scraper](https://github.com/openshift-assisted/assisted-events-scrape/blob/master/assisted-events-scrape/utils/anonymizer.py#L25)
+- Needs to expose an option for users to opt out of this feature
+- Needs to know where and how to get the remote configuration 
+
+
+### Additional Changes
+
+- Documentation about how to opt out
+- Documentation of requirements: non-disconnected environment or local s3 storage?
+- Host a remote configuration (using a CDN)
+
+### Remote Configuration
+
+The remote configuration will specify configuration data for the assisted-service such as `DATA_COLLECTION_URL`. This can be cached in a CDN.
+
+The assisted service will get this remote configuration:
+
+1. When the assisted service initially starts
+2. Every time the assisted service will send data it'll get this configuration first
+
+If the assisted service can't get the latest configuration, it'll fallback to the last configuration received or the hard-coded default it was deployed with.
+
+#### Benefits
```

Comment:
This has me thinking that maybe we want a way to control the granularity of the data we send?
It can be the same endpoint but maybe we want a way to tell the service that we don't need host inventory or something if we end up sending too much. That's just an example, but hopefully it illustrates the idea.

Reply:
Great idea! Added this to the future enhancement as an idea of what the remote configuration could contain!